index.js

```diff
@@ -51,5 +51,155 @@ const state = {
       price: 0.35
     }
   ],
-  cart: []
-};
\ No newline at end of file
+  cart: [],
+  total: [],
+};
+
+// Function to add item to cart
+function addToCart(index) {
+  
+  //new item
+  const newItem = state.items[index];
+
+  //check if item already exists
+  const existingItemIndex = state.cart.findIndex(item => item.id === newItem.id);
+
+  if (existingItemIndex !== -1) 
+  { //increase quantity if item is already in cart
+    state.cart[existingItemIndex].quantity++;
+  } else
+  { //otherwise add to cart with quantity = 1
+    newItem.quantity = 1;
+    state.cart.push(newItem);
+  }
+
+  //increase total price
+  state.total.push(newItem.price);
+  const sum = state.total.reduce((total, currentValue) => 
+  total + currentValue, 0);
+  renderCart() //update cart
+}
+
+//function to increase quantity
+function increaseQuantity(item) {
+  item.quantity++;
+  state.total.push(item.price);
+  renderCart(); //update the cart display
+}
+
+//function to decrease quantity
+function decreaseQuantity(item) {
+  if (item.quantity === 1)
+  {
+    state.cart.pop(item);
+    state.total.pop(item.price);
+  } else
+
+  {
+    item.quantity--;
+    state.total.pop(item.price);
+  }
+  renderCart();//update cart
+}
+
+//loop through the itemsList
+for (let i = 0; i < state.items.length; i++)
+{
+  //create new item
+  const storeItem = document.createElement("li");
+
+  //create div for store item icon
+  const storeItemDiv = document.createElement("div");
+  storeItemDiv.classList.add("store--item-icon")
+
+  //create img element for the icon
+  const storeItemImg = document.createElement("img");
+  storeItemImg.src = "assets/icons/" + state.items[i].id + ".svg";
+  storeItemImg.alt = state.items[i].name; //loop
+
+  //append the img element to div
+  storeItemDiv.appendChild(storeItemImg);
+
+  //create button
+  const addButton = document.createElement("button");
+  addButton.textContent = "Add to cart";
+
+  // Add event listener to "Add to cart" button
+  addButton.addEventListener("click", () => {
+    addToCart(i);
+  });
+
+  //append icon div and button to item
+  storeItem.appendChild(storeItemDiv);
+  storeItem.appendChild(addButton);
+
+  //find the element to put storeItem
+  var itemList = document.querySelector(".item-list.store--item-list");
+
+  //append the list item to the item list
+  itemList.appendChild(storeItem);
+}
+
+//function to render cart
+function renderCart() {
+  // Find the cart item list
+  const cartItemList = document.querySelector(".item-list.cart--item-list");
+  
+  // Clear the existing content of the cart
+  cartItemList.innerHTML = "";
+
+  //set total to equal 0 if no item is in the cart
+  const total = 0;
```

Comment:
you've set the total variable to 0 every time you render the cart, which negates the total calculation. This should be removed, and the total should be calculated based on the cart contents

Reply:
I have refactored, with this feedback in mind